### PR TITLE
Refactor: Add extern (D) overloadApply and use lambda

### PR DESCRIPTION
**Pull request by 9rnsr**

### CODE REVIEW

File: src/func.d [hunk 1]
```diff
@@ -3847,29 +3813,43 @@ extern (C++) int overloadApply(Dsymbol fstart, void* param, int function(void*,
...
+        {
+            d.error("is aliased to a function");
+            break;
+            // BUG: should print error message?
+        }
     }
     return 0;
 }
 
+extern (D) int overloadApply(Dsymbol fstart, scope int delegate(Dsymbol) dg)
```
Comment (yebblies) [inline src/func.d]: IIUC the D version is wrapping the C++ version.  Instead, could the extern(C++) version wrap the D version?

Comment (9rnsr) [inline src/func.d]: Sounds better. Done.

**Comment by yebblies** [comment]
This would have been a lot easier to review, as usual, if you hadn't mixed in 1000 other refactoring changes at the same time.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**